x/ecocredit/server/marketplace/prune_test.go

```diff
@@ -66,3 +66,41 @@ func TestSell_Prune(t *testing.T) {
 	_, err = s.marketStore.SellOrderTable().Get(s.ctx, shouldBeValid)
 	assert.NilError(t, err)
 }
+
+// TestPrune_NilExpiration tests that sell orders with nil expirations are not deleted when PruneOrders is called.
+func TestPrune_NilExpiration(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	s.testSellSetup(batchDenom, ask.Denom, ask.Denom[1:], "C01", start, end, creditType)
+
+	blockTime, err := time.Parse("2006-01-02", "2020-01-01")
+	assert.NilError(t, err)
+	expired, err := time.Parse("2006-01-02", "2010-01-01")
+	assert.NilError(t, err)
```

Comment:
We recently added `types.ParseDate` and should probably make use of it here. No need to do in the above test as it looks like it will get updated in https://github.com/regen-network/regen-ledger/pull/1118.

```suggestion
	blockTime, err := types.ParseDate("block time", "2020-01-01")
	assert.NilError(t, err)
	expired, err := time.Parse("expiration", "2010-01-01")
	assert.NilError(t, err)
```